api/src/db-connection.ts

```diff
@@ -112,6 +112,31 @@ export class Connection {
     return isNaN(version) ? null : version;
   }
 
+  async getLatestProcessedArxivVersion(paperSelector: PaperSelector): Promise<number> {
+    if (isS2Selector(paperSelector)) {
+      // TODO (mjlangan): Error, or maybe null, think about this
```

Comment:
@mjlangan would it be helpful to discuss this?

Reply:
Sure. Promises can return either a success or a failure, so that's where the "maybe return an error" thought comes from (it also gives the opportunity to return some data about why it failed). Returning null makes it more consistent with the other functions in here. Since this is returning just a `number` type, and that number value has to be 0 or larger, `-1` is also a reasonable value to return when there's no data.

I think it seems fine to return null in this case for consistency, but we might want to look into failing the promise with some kind of error object so the caller has more context about why the request failed.

Reply:
Agreed! All three of these ideas (-1, null, and error) sound suitable to me. Based on your discussion, I do have a slight preference for throwing an error to keep with conventions of promises (i.e., calling a `reject` callback with details).

You're also right that `getLatestPaperDataVersion` is another async function that returns null. It makes me think maybe we should migrate that method to throw an error at some point too. I can't remember my rationale behind having it return null.

Reply:
I'll go with returning `null` to keep it consistent with the rest of the file, and we can revisit with thrown errors in a subsequent change.